packages/teleterm/src/ui/services/quickInput/types.ts

```diff
@@ -31,16 +31,35 @@ export type AutocompleteToken = {
   startIndex: number;
 };
 
-type AutocompleteBase<T> = {
+type AutocompleteResultBase<T> = {
   kind: T;
+  // Command includes the result of parsing whatever was parsed so far.
+  // This means that in case of `tsh ssh roo`, the command will say that we want to launch `tsh ssh`
+  // with `roo` as `loginHost`.
+  command: AutocompleteCommand;
 };
 
 export type AutocompletePartialMatch =
-  AutocompleteBase<'autocomplete.partial-match'> & {
+  AutocompleteResultBase<'autocomplete.partial-match'> & {
     suggestions: Suggestion[];
     targetToken: AutocompleteToken;
   };
 
-export type AutocompleteNoMatch = AutocompleteBase<'autocomplete.no-match'>;
+export type AutocompleteNoMatch =
+  AutocompleteResultBase<'autocomplete.no-match'>;
 
 export type AutocompleteResult = AutocompletePartialMatch | AutocompleteNoMatch;
+
+type CommandBase<T> = {
+  kind: T;
+};
+
+export type AutocompleteUnknownCommand = CommandBase<'command.unknown'>;
+
+export type AutocompleteTshSshCommand = CommandBase<'command.tsh-ssh'> & {
+  loginHost: string;
```

Comment:
This version doesn't support passing arbitrary options yet, I think I'll work on that today.